### PR TITLE
Extract manifest modification functions from playlist loaders to manifest module

### PR DESCRIPTION
**Pull request by gesinger**
## Description

After parsing an HLS or DASH manifest string via m3u8-parser or mpd-parser, the respective playlist loader (playlist-loader or dash-playlist-loader) would modify the manifest objects to add properties or change their structure so that VHS could use them as if they were standardized.

The functions for doing this lived in playlist-loader and dash-playlist-loader, however, with the upcoming possibility of passing in a vhs-json object as the source (representing a manifest string already parsed by m3u8-parser/mpd-parser), the functions needed to be extracted to be reused appropriately, in a way that could act on either HLS or DASH playlists in the same fashion.

This extracts that functionality to the manifests.js module.

## Requirements Checklist
- [X] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [X] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://jsbin.com/gejugat/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/manifest.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+      properties.playlists[0].uri = phonyUri;
+      properties.playlists[0].id = id;
+      // set up ID references
+      master.playlists[id] = properties.playlists[0];
+    }
+  });
+
+  setupMediaPlaylists(master);
+  resolveMediaGroupUris(master);
```
Comment (brandonocasey) [inline src/manifest.js]: Should we return master here? Seems like the function this was based on used to do that.

File: src/manifest.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      'VIDEO': {},
+      'CLOSED-CAPTIONS': {},
+      'SUBTITLES': {}
+    },
+    uri: window.location.href,
+    playlists: [{
+      uri,
+      id,
+      resolvedUri: uri,
+      attributes: {}
```
Comment (brandonocasey) [inline src/manifest.js]: Should we add back some of the old inline comments from the code this function is based on?

File: src/manifest.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+      // throughout VHS, but some formats (e.g., DASH) don't have external URIs
+      // TODO: consider adding dummy URIs in mpd-parser
+      const phonyUri = `placeholder-uri-${i}`;
+
+      master.playlists[i].uri = phonyUri;
+    }
+  }
+
+  forEachMediaGroup(master, (properties, mediaType, groupKey, labelKey) => {
+    if (properties.playlists &&
```
Comment (brandonocasey) [inline src/manifest.js]: Should we return here rather than doing it in the `if` scope?

File: src/manifest.js (hunk 1 above)
Comment (gesinger) [reply src/manifest.js]: Returning master would make sense, and I can definitely add it, but where did you see it used to be returned?

Comment (brandonocasey) [reply src/manifest.js]: Looking back parseXml returned it and since it still does I guess that we don't need to do it here.

**Pull request merged**